src/main/java/org/gentoo/java/ebuilder/maven/MavenCache.java

```
@@ -43,13 +43,13 @@ public String getDependency(final String groupId, final String artifactId,
         final Map<String, List<CacheItem>> artifactIds = cache.get(groupId);
 
         if (artifactIds == null) {
-            return null;
+            return "!!!groupId-not-found!!!";
```

both these changes affect the behavior of MavenEbuilder. when it receives null value, it skips the dependency in dependencies declaration and mentions that in related comments where are dependencies that were impossible to resolve are mentioned. when returning string instead of null, the returned string will be placed in the dependencies without mentioning this fact in the related comment which makes the result ambiguous. i suggest you modify MavenEbuilder instead to enhance the info reported in the comments with information for what reason the match failed (if that is ever needed). you can enhance MavenCache for that to be able to query the cache for specific groupId or groupId:artifactId coordinates, something like hasGroupId(groupId) and hasArtifactId(groupId, artifactId).

---

> it skips the dependency in dependencies declaration and mentions that in related comments where are dependencies that were impossible to resolve are mentioned. 

This is not true.  If it results in null, the dependency is silently dropped at https://github.com/gentoo/java-ebuilder/blob/master/src/main/java/org/gentoo/java/ebuilder/maven/MavenEbuilder.java#L206 without any comments.

That's the motivation of me to insert a place holder.

> the returned string will be placed in the dependencies without mentioning this fact in the related comment which makes the result ambiguous. 

The comments are like
`# org.fusesource.leveldbjni:leveldbjni-all:1.8 -> !!!groupid-not-found!!!`
cannot say it is ambiguous.

If we care about the missing dependencies, the ebuild is broken anyway.  Package the dependencies.   If not, we could remove !!!.*!!! afterwards anyway.